capacity4more/modules/c4m/restful/c4m_restful_quick_post/components/c4m-app/src/directives/list-terms/list-terms.js

```diff
@@ -76,17 +75,14 @@ angular.module('c4mApp')
                 }
               });
             }
-          });
-
-          if (selectedCount > 3) {
-            angular.element("#" + scope.type + "_description").addClass('tooMany');
-            if (scope.popup) {
-              scope.popup = 0;
+            // Don't check if selected more than 3 topics or regions.
+            if (scope.type == "topic" || scope.type == "geo") {
+              if (selectedCount > 3) {
+                scope.model[vocab.id] = false;
+                angular.element("#" + scope.type + "_description").addClass('tooMany');
+              }
             }
-          }
-          else {
```

Comment:
@amitaibu With new logic i don't see necessity in this class. Because it's never be tooMany because maximum 3 ^^